### PR TITLE
Revised retrieve local categories

### PR DESCRIPTION
**Pull request by mjohnsullivan**
Revised the _retrieveLocalCategories() method to simplify it down some. Also added a streaming version of the method, but it's more complicated than the revised, so it might be best to stick with the non-streams version.

### CODE REVIEW

**Comment by maryx** [comment]
Not sure why but I'm getting the perpetual loading indicator screen, even with `flutter run --preview-dart-2`. Comes with error:
```
E/flutter (27766): [ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
E/flutter (27766): type '(Object) => dynamic' is not a subtype of type '(String) => Future'
```

File: unit_converter/unit_converter/lib/category_route.dart [hunk 1]
```diff
@@ -110,26 +110,42 @@ class _CategoryRouteState extends State<CategoryRoute> {
 
   /// Retrieves a list of [Categories] and their [Unit]s
   Future<Null> _retrieveLocalCategories() async {
-    final json =
-        DefaultAssetBundle.of(context).loadString('assets/regular_units.json');
-    final decoder = JsonDecoder();
-    final data = decoder.convert(await json);
+    final data = await
+        DefaultAssetBundle.of(context).loadStructuredData('assets/regular_units.json', new JsonDecoder().convert);
```
Comment (maryx) [inline unit_converter/unit_converter/lib/category_route.dart]: Maybe this isn't supposed to be a `new` JsonDecoder?

**Comment by mjohnsullivan** [comment]
New is now optional :)

**Comment by maryx** [comment]
Right, so should we remove the word `new` in `new JsonDecoder` to avoid confusion?

Also, are you getting an unhandled exception error when you run this branch?

**Comment by maryx** [comment]
@mjohnsullivan Please resolve conflicts and address comments.

(Alternatively, I can take over this PR if you're short on time).

**Comment by mjohnsullivan** [comment]
Resolved conflict

**Comment by mjohnsullivan** [comment]
This got messy; closing this PR down and will create a new one

**Pull request closed without merging**